docs/source/stability_policy.rst

```diff
@@ -0,0 +1,132 @@
+Stability Policy
+================
+
+The ``python-telegram-bot`` library (PTB) tries to provides a stable interface to its users.
+As both the user base and the complexity of the library has grown significantly since PTBs conception, we introduced below explicit stability Policy with version NEXT.VERSION.
+
+Large parts of the :mod:`telegram` package are the Python representations of the Telegram Bot API, the stability policy of which PTB can not influence.
+This policy hence includes some special cases for those parts.
+
+What does this policy cover?
+----------------------------
+
+This policy includes any API or behavior that is documented in this documentation.
+This covers both the :mod:`telegram` package and the :mod:`telegram.ext` package.
+
+What doesn't this policy cover?
+-------------------------------
+
+Introduction of new features or changes of flavors of comparable behavior (e.g. the default for the used HTTP protocol version) are not covered by this policy.
+The internal structure of classes in PTB, i.e. things like the result of ``dir(obj))`` or the contents of ``obj.__dict__``, may change.
+
+Objects are in general not guaranteed to be pickleable (unless stated otherwise) and pickled objects from one version of PTB may not be loadable in future versions.
+We may provide a way to convert pickled objects from one version to another, but this is not guaranteed.
+
+Functionality that is part of PTBs API but is explicitly documented to not be intended to be used directly by users (e.g. :meth:`telegram.request.BaseRequest.do_request`) may change.
+This also applies to functions or attributes marked as final in the sense of `PEP 591 <https://www.python.org/dev/peps/pep-0591/>`__.
+
+PTB has dependencies to 3rd party packages.
+The versions that PTB uses of these 3rd party packages may change if that does not change PTBs public API.
+
+PTB does not give guarantees about which Python versions are supported.
+In general, we will try to support all Python versions that have not yet reached their end of life, but we reserve ourselves the option to drop support for Python versions earlier if that benefits the advancement of the library.
+
+.. _bot-api-functionality-1:
+
+Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~
+
+Comparison of equality of the classe in the :mod:`telegram` package is subject to change and the PTB team will update the behavior to best reflect updates in the Bot API.
```

Comment:
so we will immediately update those now, meaning a class from the previous version (e.g. saved in chat_data) compared to the same class in a new version will not be the same? I remember us not having a consent on this, right? In any case, I would expand on this like I did in the beginning to make sure people understand what happen when they update.

Reply:
Correct. So far we did not touch the equality comparison on non-breaking releases. I agree however that it's nicer to have that in sync with the API and with a documented policy in place I'm good with doing that.

What exactly would you like to elaborate?

Reply:
_and the PTB team will update the behavior to best reflect updates in the Bot API. This can mean that comparing an object with one from a previous version (e.g. when loading from storage) can fail. We will warn about this in the affected object documentation._

I am not happy about the wording, but the gist: If we need to update ._id_attr, we will warn in the version string, but it can mean a fail in edge cases.